src/components/Dialog/Dialog.jsx

```diff
@@ -346,12 +348,19 @@ export default class Dialog extends PureComponent {
               ...modifiers
             ]}
           >
-            {({ placement, style, ref, arrowProps }) => {
+            {({ placement, style, ref, arrowProps, isReferenceHidden }) => {
               if (!this.isShown() && placement) {
                 return null;
               }
+
+              if (hideWhenReferenceHidden && isReferenceHidden) {
+                const event = new CustomEvent("onReferenceHidden");
+                this.hideDialog(event, "onReferenceHidden");
+              }
+
               return (
                 <DialogContent
+                  isReferenceHidden={hideWhenReferenceHidden && isReferenceHidden}
```

Comment:
maybe send isReferenceHidden as is
and also pass the other prop (hideWhenReferenceHidden) to the dialogContent component?